manual/manual_compiler.rst

```diff
@@ -817,3 +817,198 @@ This measurement partitioning is built into the :py:meth:`get_operator_expectati
+
+When the above circuit is run from an all-zero state, the Y and CX gates at the
+beginning just have the effect of putting both qubits in the "1" state (ignoring
+unobservable global phase), so they could be replaced with two X gates. This is
+exactly what the :py:meth:`SimplifyInitial` pass does.
+
+.. jupyter-execute::
+
+    from pytket.passes import SimplifyInitial
+
+    SimplifyInitial().apply(c)
+    print(c.get_commands())
+
+This pass tracks the state of qubits known to be initialised to zero forward
+through the circuit, for as long as the qubits remain in a computational basis
+state, either removing gates (when they don't change the state) or replacing
+them with X gates (when they invert the state).
+
+By default, this pass also replaces Measure operations acting on qubits with a
+known state by classical set-bits operations on the target bits:
+
+.. jupyter-execute::
+
+    c = Circuit(1).X(0).measure_all()
+    c.qubit_create_all()
+    SimplifyInitial().apply(c)
+    print(c.get_commands())
+
+The measurement has disappeared, replaced with a classical operation on its
+target bit. To disable this behaviour, pass the ``allow_classical=False``
+argument to :py:meth:`SimplifyInitial` when constructing the pass.
+
+Note that :py:meth:`SimplifyInitial` does not automatically cancel successive
+pairs of X gates introduced by the simplification. It is a good idea to follow
+it with a :py:meth:`RemoveRedundancies` pass in order to perform these
+cancellations.
+
+Removal of discarded operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An operation that has no quantum or classical output in its causal future has no
+effect and can be removed from the circuit. By marking a qubit as discarded, we
+tell the compiler that it has no quantum output, potentially enabling this
+simplification.
+
+Note that if the qubit is measured, even if it is then discarded, the Measure
+operation has a classical output in its causal future so will not be removed.
+
+.. jupyter-execute::
+
+    from pytket.circuit import Qubit
+    from pytket.passes import RemoveDiscarded
+
+    c = Circuit(3, 2)
+    c.H(0).H(1).H(2).CX(0, 1).Measure(0, 0).Measure(1, 1).H(0).H(1)
+    c.qubit_discard(Qubit(0))
+    c.qubit_discard(Qubit(2))
+    RemoveDiscarded().apply(c)
+    print(c.get_commands())
+
+The Hadamard gate following the measurement on qubit 0, as well as the Hadamard
+on qubit 2, have disappeared, because those qubits were discarded. The Hadamard
+following the measurement on qubit 1 remains, because that qubit was not
+discarded.
+
+Commutation of measured classical maps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The last type of contextual optimization is a little more subtle. Let's call a
+quantum unitary operation a `classical map` if it sends every computational
+basis state to a computational basis state, possibly composed with a diagonal
+operator. For example, X, Y, Z, Rz, CX, CY, CZ and Sycamore are classical maps,
+but Rx, Ry and H are not.
```

Comment:
This statement relies on knowledge of the unitary definitions and I wouldn't expect all readers to know what the Sycamore gate is, so it might be good to tell readers to check the API ref for `OpType` to see what each unitary looks like and if it would be amenable to this optimisation

Reply:
Done.